File: wrappers/mvc/src/Kendo.Mvc/UI/Gantt/GanttView.cs
namespace Kendo.Mvc.UI
{
    using System;
    using System.Linq;
    using System.Collections.Generic;
    using System.Web.Routing;
    using Kendo.Mvc.Extensions;

    public class GanttView : JsonObject
    {
        public GanttView()
        {
            //>> Initialization
        
        //<< Initialization
        }

        //>> Fields
        
        public bool? Selected { get; set; }
        
        public double? SlotSize { get; set; }
        
        public string TimeHeaderTemplate { get; set; }

        public string TimeHeaderTemplateId { get; set; }
        
        public string DayHeaderTemplate { get; set; }

        public string DayHeaderTemplateId { get; set; }
        
        public string WeekHeaderTemplate { get; set; }

        public string WeekHeaderTemplateId { get; set; }
        
        public string MonthHeaderTemplate { get; set; }

        public string MonthHeaderTemplateId { get; set; }
        
        public string YearHeaderTemplate { get; set; }

        public string YearHeaderTemplateId { get; set; }
        
        public string ResizeTooltipFormat { get; set; }
        
        public GanttViewType? Type { get; set; }
        
        //<< Fields

        protected override void Serialize(IDictionary<string, object> json)
        {
            //>> Serialization
        
            if (Selected.HasValue)
            {
                json["selected"] = Selected;
            }
                
            if (SlotSize.HasValue)
            {
                json["slotSize"] = SlotSize;
            }
                
            if (!string.IsNullOrEmpty(TimeHeaderTemplateId))
            {
                json["timeHeaderTemplate"] = new ClientHandlerDescriptor {
                    HandlerName = string.Format(
                        "jQuery('#{0}').html()",
                        TimeHeaderTemplateId
                    )
                };
            }
            else if (!string.IsNullOrEmpty(TimeHeaderTemplate))
            {
                json["timeHeaderTemplate"] = TimeHeaderTemplate;
            }
                
            if (!string.IsNullOrEmpty(DayHeaderTemplateId))
            {
                json["dayHeaderTemplate"] = new ClientHandlerDescriptor {
                    HandlerName = string.Format(
                        "jQuery('#{0}').html()",
                        DayHeaderTemplateId
                    )
                };
            }
            else if (!string.IsNullOrEmpty(DayHeaderTemplate))
            {
                json["dayHeaderTemplate"] = DayHeaderTemplate;
            }
                
            if (!string.IsNullOrEmpty(WeekHeaderTemplateId))
            {
                json["weekHeaderTemplate"] = new ClientHandlerDescriptor {
                    HandlerName = string.Format(
                        "jQuery('#{0}').html()",
                        WeekHeaderTemplateId
                    )
                };
            }
            else if (!string.IsNullOrEmpty(WeekHeaderTemplate))
            {
                json["weekHeaderTemplate"] = WeekHeaderTemplate;
            }
                
            if (!string.IsNullOrEmpty(MonthHeaderTemplateId))
            {
                json["monthHeaderTemplate"] = new ClientHandlerDescriptor {
                    HandlerName = string.Format(
                        "jQuery('#{0}').html()",
                        MonthHeaderTemplateId
                    )
                };
            }
            else if (!string.IsNullOrEmpty(MonthHeaderTemplate))
            {
                json["monthHeaderTemplate"] = MonthHeaderTemplate;
            }
                
            if (!string.IsNullOrEmpty(YearHeaderTemplateId))
            {
                json["yearHeaderTemplate"] = new ClientHandlerDescriptor {
                    HandlerName = string.Format(
                        "jQuery('#{0}').html()",
                        YearHeaderTemplateId
                    )
                };
            }
            else if (!string.IsNullOrEmpty(YearHeaderTemplate))
            {
                json["yearHeaderTemplate"] = YearHeaderTemplate;
            }
                
            if (ResizeTooltipFormat.HasValue())
            {
                json["resizeTooltipFormat"] = ResizeTooltipFormat;
            }
            
            if (Type.HasValue)
            {
                json["type"] = Type;
            }
                
        //<< Serialization
        }
    }
}

File: wrappers/php/lib/Kendo/UI/GanttView.php
<?php

namespace Kendo\UI;

class GanttView extends \Kendo\SerializableObject {
//>> Properties

    /**
    * If set to true the view will be initially selected by the Gantt widget. The default selected view is "day".
    * @param boolean $value
    * @return \Kendo\UI\GanttView
    */
    public function selected($value) {
        return $this->setProperty('selected', $value);
    }

    /**
    * The size of the time slot headers. Values are treated as pixels.
    * @param float|string $value
    * @return \Kendo\UI\GanttView
    */
    public function slotSize($value) {
        return $this->setProperty('slotSize', $value);
    }

    /**
    * Sets the timeHeaderTemplate option of the GanttView.
    * The template used to render the time slots in "day" view
    * @param string $value The id of the element which represents the kendo template.
    * @return \Kendo\UI\GanttView
    */
    public function timeHeaderTemplateId($value) {
        $value = new \Kendo\Template($value);

        return $this->setProperty('timeHeaderTemplate', $value);
    }

    /**
    * Sets the timeHeaderTemplate option of the GanttView.
    * The template used to render the time slots in "day" view
    * @param string $value The template content.
    * @return \Kendo\UI\GanttView
    */
    public function timeHeaderTemplate($value) {
        return $this->setProperty('timeHeaderTemplate', $value);
    }

    /**
    * Sets the dayHeaderTemplate option of the GanttView.
    * The template used to render the day slots in "day" and "week" views.
    * @param string $value The id of the element which represents the kendo template.
    * @return \Kendo\UI\GanttView
    */
    public function dayHeaderTemplateId($value) {
        $value = new \Kendo\Template($value);

        return $this->setProperty('dayHeaderTemplate', $value);
    }

    /**
    * Sets the dayHeaderTemplate option of the GanttView.
    * The template used to render the day slots in "day" and "week" views.
    * @param string $value The template content.
    * @return \Kendo\UI\GanttView
    */
    public function dayHeaderTemplate($value) {
        return $this->setProperty('dayHeaderTemplate', $value);
    }

    /**
    * Sets the weekHeaderTemplate option of the GanttView.
    * The template used to render the week slots in "week" and "month" views.
    * @param string $value The id of the element which represents the kendo template.
    * @return \Kendo\UI\GanttView
    */
    public function weekHeaderTemplateId($value) {
        $value = new \Kendo\Template($value);

        return $this->setProperty('weekHeaderTemplate', $value);
    }

    /**
    * Sets the weekHeaderTemplate option of the GanttView.
    * The template used to render the week slots in "week" and "month" views.
    * @param string $value The template content.
    * @return \Kendo\UI\GanttView
    */
    public function weekHeaderTemplate($value) {
        return $this->setProperty('weekHeaderTemplate', $value);
    }

    /**
    * Sets the monthHeaderTemplate option of the GanttView.
    * The template used to render the month slots in "month" and "year" views.
    * @param string $value The id of the element which represents the kendo template.
    * @return \Kendo\UI\GanttView
    */
    public function monthHeaderTemplateId($value) {
        $value = new \Kendo\Template($value);

        return $this->setProperty('monthHeaderTemplate', $value);
    }

    /**
    * Sets the monthHeaderTemplate option of the GanttView.
    * The template used to render the month slots in "month" and "year" views.
    * @param string $value The template content.
    * @return \Kendo\UI\GanttView
    */
    public function monthHeaderTemplate($value) {
        return $this->setProperty('monthHeaderTemplate', $value);
    }

    /**
    * Sets the yearHeaderTemplate option of the GanttView.
    * The template used to render the year slots in "year" view.
    * @param string $value The id of the element which represents the kendo template.
    * @return \Kendo\UI\GanttView
    */
    public function yearHeaderTemplateId($value) {
        $value = new \Kendo\Template($value);

        return $this->setProperty('yearHeaderTemplate', $value);
    }

    /**
    * Sets the yearHeaderTemplate option of the GanttView.
    * The template used to render the year slots in "year" view.
    * @param string $value The template content.
    * @return \Kendo\UI\GanttView
    */
    public function yearHeaderTemplate($value) {
        return $this->setProperty('yearHeaderTemplate', $value);
    }

    /**
    * The format used to display the start and end dates in the resize tooltip.
    * @param string $value
    * @return \Kendo\UI\GanttView
    */
    public function resizeTooltipFormat($value) {
        return $this->setProperty('resizeTooltipFormat', $value);
    }

    /**
    * The view type. Supported types are "day", "week", "month" and "year".
    * @param string $value
    * @return \Kendo\UI\GanttView
    */
    public function type($value) {
        return $this->setProperty('type', $value);
    }

//<< Properties
}

?>
